src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -1206,13 +1208,13 @@ public static void Main()
 {
     // Code size       29 (0x1d)
     .maxstack  3
-    IL_0000:  ldarg.1
-    IL_0001:  brtrue.s   IL_000e
-    IL_0003:  ldstr      ""s""
-    IL_0008:  newobj     ""System.ArgumentNullException..ctor(string)""
-    IL_000d:  throw
-    IL_000e:  ldc.i4.s   -2
-    IL_0010:  newobj     ""C.<GetChars>d__0..ctor(int)""
+    IL_0000:  ldc.i4.s   -2
```

Comment:
This reordering seems incorrect. I wouldn't necessarily try to fix it in this PR, just prototype it. But we should be doing the null checks _first_ not after creating the enumerable closure object.